packages/world/src/systems/TerrainSystem.sol

```diff
@@ -4,16 +4,25 @@ pragma solidity >=0.8.24;
 import { System } from "@latticexyz/world/src/System.sol";
 import { Vec3 } from "../Vec3.sol";
 import { ExploredChunkCount } from "../codegen/tables/ExploredChunkCount.sol";
+import { RegionMerkleRoot } from "../codegen/tables/RegionMerkleRoot.sol";
 import { ExploredChunkByIndex, ExploredChunk, InitialEnergyPool, LocalEnergyPool } from "../utils/Vec3Storage.sol";
 import { SSTORE2 } from "../utils/SSTORE2.sol";
-import { INITIAL_ENERGY_PER_VEGETATION } from "../Constants.sol";
+import { INITIAL_ENERGY_PER_VEGETATION, REGION_SIZE, CHUNK_SIZE } from "../Constants.sol";
+import { MerkleProof } from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
```

Comment:
solady also has a merkle proof library, which is probably more efficient (I'm sure oz is fine too)

Reply:
would leave for a followup if you agree!